libcnb-cargo/src/cli.rs

```diff
@@ -25,6 +26,9 @@ pub(crate) struct PackageArgs {
     /// Build for the target triple
     #[arg(long, default_value = "x86_64-unknown-linux-musl")]
     pub target: String,
+    /// Directory for packaged buildpacks, defaults to 'libcnb-packaged' in Cargo workspace root
+    #[arg(long)]
+    pub package_dir: Option<PathBuf>,
```

Comment:
I'm not 💯 happy with the way the default is handled here. But since the actual path depends on the cargo metadata, we cannot determine it here. Essentially, the default path is dynamic. From my research, this is the best we can do with `clap` in such cases. Happy to get corrected :)

Reply:
would using custom [value parser](https://docs.rs/clap/latest/clap/builder/struct.ValueParser.html) be possible?

Reply:
From what I've seen, the logic around mandatory arguments and defaults is implemented outside of `ValueParser`s. Even if we could handle optional values in `ValueParsers`, we cannot specify the default before parsing so that it will show up when using `--help` (which is my main concern).